lib/obsolete.gd

```diff
@@ -479,6 +479,30 @@ DeclareObsoleteSynonym( "MutableIdentityMat", "IdentityMat", "4.8" );
 ##
 DeclareObsoleteSynonym( "MutableNullMat", "NullMat", "4.8" );
 
+#############################################################################
+##
+#F  IsSemilatticeAsSemigroup( <S> ) is the semigroup <S> a semilattice
+##
+##  <#GAPDoc Label="IsSemilatticeAsSemigroup">
+##  <ManSection>
+##  <Prop Name="IsSemilatticeAsSemigroup" Arg='S'/>
+##
+##  <Description>
+##    <C>IsSemilatticeAsSemigroup</C> returns <K>true</K> if the semigroup
+##    <A>S</A> is a semilattice and <K>false</K> if it is not. <P/>
+##
+##    A semigroup is a <E>semilattice</E> if it is commutative and every
+##    element is an idempotent. The idempotents of an inverse semigroup form a
+##    semilattice.
+##
+##    This is identical to <Ref Prop="IsSemilattice" BookName = "Semigroups"/> #
+##    and is present in &GAP;&nbsp;4.8 #  only for the sake of compatibility with
+##    beta-releases.  #  It should <E>not</E> be used in new code.
+##  </Description>
+##  </ManSection>
+##  <#/GAPDoc>
+##
+DeclareObsoleteSynonym( "IsSemilatticeAsSemigroup", "IsSemilattice", "4.8" );
```

Comment:
Right, I already prepared a new versions of Semigroups (2.7) and Smallsemi (0.6.9) so that it is compatible with all of the changes in this pull request. I mentioned this somewhere else on this page, and I followed the suggestion of @alex-konovalov to not release these versions of the packages until the pull request was accepted.